src/types/st-object.ts

```diff
@@ -1,79 +1,101 @@
 import { Field } from "../enums";
-const _ = require("lodash");
-const { BinarySerializer } = require("../serdes/binary-serializer");
-const { SerializedType } = require("./serialized-type");
-
-const STObject = makeClass(
-  {
-    mixins: SerializedType,
-    statics: {
-      fromParser(parser, hint) {
-        const end = typeof hint === "number" ? parser.pos() + hint : null;
-        const so = new this();
-        while (!parser.end(end)) {
-          const field = parser.readField();
-          if (field.name === "ObjectEndMarker") {
-            break;
-          }
-          so[field.name] = parser.readFieldValue(field);
-        }
-        return so;
-      },
-      from(value) {
-        if (value instanceof this) {
-          return value;
-        }
-        if (typeof value === "object") {
-          return _.transform(
-            value,
-            (so, val, key) => {
-              const field = Field[key];
-              if (field) {
-                so[field.name] = field.associatedType.from(val);
-              } else {
-                so[key] = val;
-              }
-            },
-            new this()
-          );
-        }
-        throw new Error(`${value} is unsupported`);
-      },
-    },
-    fieldKeys() {
-      return Object.keys(this)
-        .map((k) => Field[k])
-        .filter(Boolean);
-    },
-    toJSON() {
-      // Otherwise seemingly result will have same prototype as `this`
-      const accumulator = {}; // of only `own` properties
-      return _.transform(
-        this,
-        (result, value, key) => {
-          result[key] = value && value.toJSON ? value.toJSON() : value;
-        },
-        accumulator
-      );
-    },
-    toBytesSink(sink, filter = () => true) {
-      const serializer = new BinarySerializer(sink);
-      const fields = this.fieldKeys();
-      const sorted = _.sortBy(fields, "ordinal");
-      sorted.filter(filter).forEach((field) => {
-        const value = this[field.name];
-        if (!field.isSerialized) {
-          return;
-        }
-        serializer.writeFieldAndValue(field, value);
-        if (field.type.name === "STObject") {
-          serializer.put(Buffer.from([0xe1]));
-        }
+import { SerializedType } from "./serialized-type";
+import { BinaryParser } from "../serdes/binary-parser";
+import { BinarySerializer, BytesList } from "../serdes/binary-serializer";
+
+const OBJECT_END_MARKER = Buffer.from([0xe1]);
+
+/**
+ * Class for Serializing/Deserializing objects
+ */
+class STObject extends SerializedType {
+  /**
+   * Construct a STObject from a BinaryParser
+   *
+   * @param parser BinaryParser to read STObject from
+   * @returns A STObject object
+   */
+  static fromParser(parser: BinaryParser): STObject {
+    const list: BytesList = new BytesList();
+    const bytes: BinarySerializer = new BinarySerializer(list);
+
+    while (!parser.end()) {
+      const field = parser.readField();
+      if (field.name === "ObjectEndMarker") {
```

Comment:
optional: You could refactor these strings to be private constants in this file since you use them repeatedly.